nixpkgs/nixpkgs.bzl

```diff
@@ -101,3 +103,68 @@ nixpkgs_package = repository_rule(
   local = True,
   environ = ["NIX_PATH"],
 )
+
+def _cc_configure_custom(ctx):
+  overriden_tools = {
+    "gcc": ctx.path(ctx.attr.gcc),
+    "ld": ctx.path(ctx.attr.ld),
+  }
+  return cc_autoconf_impl(ctx, overriden_tools)
+
+cc_configure_custom = repository_rule(
+  implementation = _cc_configure_custom,
+  attrs = {
+    "gcc": attr.label(
+      executable=True,
+      cfg="host",
+      allow_single_file=True,
+      doc="`gcc` to use in cc toolchain",
+    ),
+    "ld": attr.label(
+      executable=True,
+      cfg="host",
+      allow_single_file=True,
+      doc="`ld` to use in cc toolchain",
+    ),
+  },
+  local = True,
+  environ = [
+        "ABI_LIBC_VERSION",
```

Comment:
Where does this list of environment variables come from?

Reply:
The original rule [here](https://github.com/bazelbuild/bazel/blob/master/tools/cpp/cc_configure.bzl#L58) depends on these env vars. Since we're re-using implementation here, it makes sense that our rule also necessarily depends on the same set of variables.

Reply:
Yes but if we make a nixpkgs specific rule, then by dropping *all* these env variables we make the build more hermetic, and avoid getting subtle bugs down the line because of a mismatch in the env vars we import vs those the upstream rule imports. Cross compilation is not supported by this rule in any case.